server/names/common.go

```diff
@@ -17,31 +17,29 @@ package names
 import (
 	"fmt"
 	"regexp"
+
+	"github.com/google/uuid"
 )
 
-// We might extend this to all characters that do not require escaping.
-// See "Resource ID Segments" in https://aip.dev/122.
-const NameRegex = "([a-zA-Z0-9-_\\.]+)"
+// The format of a resource identifier.
+// This may be extended to include all characters that do not require escaping.
+// See https://aip.dev/122#resource-id-segments.
+const identifier = "([a-zA-Z0-9-_\\.]+)"
 
-// Generated revision names are lowercase hex strings, but we also
-// allow user-specified revision tags which can be mixed-case strings
-// containing dashes.
-const RevisionRegex = "(@[a-zA-z0-9-]+)?"
+// The format of a custom revision tag.
+const revisionTag = "(@[a-zA-z0-9-]+)?"
 
-func ValidateID(id string) error {
-	r := regexp.MustCompile("^" + NameRegex + "$")
-	m := r.FindStringSubmatch(id)
-	if m == nil {
-		return fmt.Errorf("invalid id '%s'", id)
-	}
-	return nil
+// GenerateID generates a random resource ID.
+func GenerateID() string {
+	return uuid.New().String()[:8]
```

Comment:
What happens if this produces an ID that's already in use? Given the low likelihood, I think it would be OK if the request failed and had to be retried but not OK if the request proceeded and an existing resource was clobbered. Could we verify this behavior by testing with shorter IDs?

Reply:
We might retry ourselves (in the API request hander) since we know when we are using auto-generated IDs.

Reply:
That depends on the implementation. Currently it results in an "already exists" error which might be confusing in the unlikely case where it happens. Implementing retry would eliminate the issue.

Writing an automated test for this would require some refactoring to allow injection of a custom randomness source at server create time. Let me know if you want me to proceed with that.

Reply:
Let's discuss it on PR#70 - I don't think this is an immediate priority but we'll probably want to eventually address it